WakaTime/Settings.swift

```diff
@@ -1,41 +1,85 @@
 
     var body: some View {
-        GeometryReader { _ in
-            VStack {
-                Text("WakaTime API Key:")
-                TextField("apikey", text: $apiKey)
-                HStack {
-                    Button("Save") {
-                        ConfigFile.setSetting(section: "settings", key: "api_key", val: $apiKey.wrappedValue)
-                        dismiss()
-                    }
-                    .keyboardShortcut(.defaultAction)
-                    Button("Cancel") {
-                        dismiss()
+        GeometryReader { geometry in
+            HStack {
+                VStack(alignment: .leading) {
+                    Toggle("Launch on login", isOn: $isToggleOn)
+                        .onChange(of: isToggleOn) { newValue in
+                            PropertiesManager.shouldLaunchOnLogin = newValue
+                            if newValue {
+                                registerAsLoginItem()
+                            } else {
+                                unregisterAsLoginItem()
+                            }
+                        }
+                    Spacer()
+                    Divider()
+                    Spacer()
+                    Text("WakaTime API Key:")
+                    TextField("apikey", text: $apiKey).frame(width: 200)
+                    HStack {
+                        Button("Save") {
+                            ConfigFile.setSetting(section: "settings", key: "api_key", val: $apiKey.wrappedValue)
+                            dismiss()
+                        }
+                        .keyboardShortcut(.defaultAction)
+                        Button("Cancel") {
+                            dismiss()
+                        }
+                        .keyboardShortcut(.cancelAction)
                     }
-                    .keyboardShortcut(.cancelAction)
+                    Spacer()
+                    Text("Version: \(Bundle.main.version) (\(Bundle.main.build))")
+                        .padding(.top)
                 }
-                Text("Version: \(Bundle.main.version) (\(Bundle.main.build))")
-                    .padding()
+                .padding()
+                .frame(maxWidth: geometry.size.width)
             }
-            .padding()
         }
         .task {
             if let window = NSApplication.shared.windows.first(where: { $0.identifier?.rawValue == "settings" }) {
                 window.styleMask.insert(.closable)
                 window.styleMask.remove(.miniaturizable)
                 window.styleMask.remove(.resizable)
-                window.setFrame(NSRect(x: window.frame.origin.x, y: window.frame.origin.y, width: 400, height: 140), display: true)
+                window.setFrame(NSRect(x: window.frame.origin.x, y: window.frame.origin.y, width: 400, height: 180), display: true)
             }
         }
     }
+
+    private func registerAsLoginItem() {
+        guard
```

Comment:
Move checking preferences outside this func, since it's only necessary on startup not after changing the toggle.